python/ray/tune/execution/ray_trial_executor.py

```diff
@@ -425,46 +415,9 @@ def _setup_remote_runner(self, trial):
         # since we don't know where the remote runner is placed.
         trial.set_location(_Location())
         logger.debug("Trial %s: Setting up new remote runner.", trial)
-        # Logging for trials is handled centrally by TrialRunner, so
-        # configure the remote runner to use a noop-logger.
-        trial_config = copy.deepcopy(trial.config)
-        trial_config[TRIAL_INFO] = _TrialInfo(trial)
-        stdout_file, stderr_file = trial.log_to_file
-        trial_config[STDOUT_FILE] = stdout_file
-        trial_config[STDERR_FILE] = stderr_file
-        kwargs = {
-            "config": trial_config,
-            "logger_creator": logger_creator,
-        }
-        if trial.uses_cloud_checkpointing:
-            # We keep these kwargs separate for backwards compatibility
-            # with trainables that don't provide these keyword arguments
-            kwargs["remote_checkpoint_dir"] = trial.remote_checkpoint_dir
-            kwargs["sync_config"] = trial.sync_config
-
-            if self._trainable_kwargs:
-                kwargs.update(self._trainable_kwargs)
-
-            # Throw a meaningful error if trainable does not use the
-            # new API
-            sig = inspect.signature(trial.get_trainable_cls())
-            try:
-                sig.bind_partial(**kwargs)
-            except Exception as e:
-                raise RuntimeError(
-                    "Your trainable class does not accept a "
-                    "`remote_checkpoint_dir` or `syncer` argument "
-                    "in its constructor, but you've passed a "
-                    "`upload_dir` to your SyncConfig. Without accepting "
-                    "these parameters and passing them to the base trainable "
-                    "constructor in the init call, cloud checkpointing is "
-                    "effectively disabled. To resolve this issue, add the "
-                    "parameters to your trainable class constructor or "
-                    "disable cloud checkpointing by setting `upload_dir=None`."
-                ) from e
```

Comment:
Do we need to still include this? This is for users subclassing Trainable and providing a new constructor that doesn't take in these two args.

Reply:
We introduced this check in Ray 1.9.0 as we changed the `Trainable` interface back then. I think it's fine to remove this check now.